### ISSUE TITLE
Add Python section

### ISSUE THREAD

**Issue by kadler** [IBM/ibmi-oss-docs#20]
Useful pages
- Supported Python versions
- Using alternatives to pick the default python/python3 binary
- Using virtualenvs to install packages
- How to build native packages (install python3-devel, development tools, etc)
- Pages on important packages to install via pip: pandas, etc

**Comment by ThePrez** [comment]
FYI, I added [a Python page](https://ibmi-oss-docs.readthedocs.io/en/latest/python/README.html) a while ago that covers some, but not all of Kevin's suggestions. 
We should keep this issue open to track the filling of those gaps. Renamed accordingly. 
@abmusse , is this still something you'd like to own?

**Comment by abmusse** [comment]
Yes I can own this one and work on implementing Kevin's suggestions

**Comment by jkyeung** [comment]
Is there any style guide or even informal conventions? I have proofreader-level nitpickiness and occasionally have an urge to clean things up.

Example: I notice that there are several small headings (such as "Important Note") styled using boldface. In the source they are on a separate line, suggesting that the intention was for them to render on a separate line. But of course they don't, so it looks kind of kludgy. If indeed they're meant to be on a separate line, I can think of a few possibilities, not limited to

1. Just add another newline after the heading.
2. Add a `<p>` after the heading.
3. Use level-4 heading (`####`) instead of bold.

All these render the same in my browser, including a good deal of space between the heading and the text that follows. If that's too much separation, then my suggestion would be to keep the "heading" on the same line as the text, but set off with a colon.

So, either

> #### Important Note
> Performing a `yum upgrade` operation...

or

> **Important Note:** Performing a `yum upgrade` operation...

or maybe even

> *Important Note:* Performing a `yum upgrade` operation...

What do people think?

(There are also a lot of smaller changes I would make, mostly for consistency.)

**Comment by kadler** [comment]
I think this looks the best: **Important Note**: _Performing a yum upgrade operation..._

**Comment by kadler** [comment]
Alternatively, one could escape out to ReStructured Text and make an actual information box like the Note here: https://python-itoolkit.readthedocs.io/en/latest/api.html#itoolkit.transport.DirectTransport.close

```
` ` `eval_rst
.. note::
   Performing a yum upgrade operation ...
` ` `
```

Though that's probably overkill for this.

**Comment by kadler** [comment]
We should also something for when users get a mismatch between SSH and QSH/QP2TERM.

- If OPS is still installed, they'll get Python 3.4 which usually gives an error that it can't find its dependencies since they were installed in a different Python version or gives a syntax error since they're running Python 3.6+ code on Python 3.4 (f-strings, etc)
- If OPS is _not installed_, then they'll just get a command not found error.

Either way, the answer is to set the PATH, but the symptoms -- and thus searching to find the answer -- are different.

**Comment by kadler** [comment]
We should probably document how to use GCC 10 on IBM i 7.4 and up, since GCC 6 has header file compatibility issues.

**Comment by jkyeung** [comment]
Sorry for being kind of off-topic, but how the heck do you satisfy the DCO check from the GitHub web interface? Is it even possible? Is there some place I can set all my commits _done completely online_ to automatically include the signed-off-by line?

**Comment by jkyeung** [comment]
Thanks for the response, @abmusse - for now I am OK with just manually entering the signed-off-by line. I finally was able to pass the DCO check by using my user name (I had been trying my real name!) in the extended commit description.

**Comment by kadler** [comment]
> Sorry for being kind of off-topic, but how the heck do you satisfy the DCO check from the GitHub web interface? Is it even possible? Is there some place I can set all my commits _done completely online_ to automatically include the signed-off-by line?

> It requires all commit messages to contain the Signed-off-by line with an email address that matches the commit author.

https://github.com/dcoapp/app?tab=readme-ov-file#probot-dco

Your sign off must match the commit author. Looking at the commit in your last failed PR: https://github.com/IBM/ibmi-oss-docs/commit/58d36e919d517234bfc4d38148a463ba4fe11b37.patch

```
From: jkyeung <gallium.arsenide@gmail.com>
...
Signed-off-by: John Yeung <gallium.arsenide@gmail.com>
```

These two lines should match. Now that you have done that, your current PR is good.
